src/Compilers/Test/Core/Compilation/ControlFlowGraphVerifier.cs

```diff
@@ -390,7 +395,7 @@ void verifyCaptures()
                                 {
                                     foreach (CaptureId id in region.CaptureIds)
                                     {
-                                        Assert.True(currentState.Contains(id), $"Backward branch from [{getBlockId(predecessor.Source)}] to [{getBlockId(block)}] before capture [{id.Value}] is initialized.");
+                                        Assert.True(currentState.Contains(id), $"Backward branch from [{getBlockId(predecessor.Source)}] to [{getBlockId(block)}] before capture [{id.Value}] is initialized.\n{finalGraph}");
```

Comment:
>{finalGraph} [](start = 207, length = 12)

We are always building a string that could be quite long. Is there a version of Assert that can do this lazily? Perhaps we should consider checking condition explicitly and building the string only for failures. Same for other similar places where an Assert condition is not a constant.